syntax = "proto3";

package rtabmap.proto.RawData;

import "Time.proto";
import "Transform.proto";
import "SensorData.proto";
import "Trajectory.proto";

message MetaData
{
    int32 version = 1;
}

message InputData
{
    Transform local_pose = 1;  // might be empty
    Transform global_pose = 2;
    SensorData sensor_data = 3;
    Time time = 4;
    Transform from_updated_pose = 5;
    bool temporary = 6;
}

message UpdatePosesData
{
    Trajectories trajectories = 1;
    Transform global_to_local = 2;  // might be empty
    Time skip_local_maps_upto = 3;  // might be empty
}

message RawData
{
    oneof raw_data
    {
        InputData input_data = 1;
        UpdatePosesData update_poses_data = 2;
    }
}
